Orange/classification/knn.py

```diff
@@ -11,5 +11,7 @@ class KNNLearner(SklLearner):
     def __init__(self, n_neighbors=5, metric="euclidean", weights="uniform",
                  algorithm='auto',
                  preprocessors=None):
+        if metric == "mahalanobis":
+            algorithm = "brute"
```

Comment:
Please add a comment explaining why this is necessary. For the record, this works for me:
```py
>>> from sklearn.neighbors import KNeighborsClassifier
>>> nn = KNeighborsClassifier(metric='mahalanobis', algorithm='auto')
>>> X = np.random.random((10, 2))
>>> y = (np.random.random(10) > .5).astype(int)
>>> nn.fit(X, y)
>>> nn.predict(X)
array([1, 1, 1, 1, 0, 0, 0, 0, 0, 0])
```